longevity_oos_test.py

```diff
@@ -0,0 +1,365 @@
+        for node in self.db_cluster.nodes:
+            wait_no_tablets_migration_running(node, timeout=7200)
+
+    def get_compactions(self, node: BaseNode, interval: int) -> int:
+        """
+        Get the number of compactions that occurred on the node during the specified interval.
+
+        :param node: The node to get the compactions for.
+        :param interval: The time interval for the query, as last X seconds.
+        :return: The number of compactions that occurred on the node during the specified interval.
+        """
+        compaction_query = f'sum(scylla_compaction_manager_compactions{{instance=~".*?{node.private_ip_address}.*?"}})'
+        now = time()
+        results = self.prometheus_db.query(query=compaction_query, start=now - interval, end=now)
+        self.log.info(f"Compactions on node {node.name} ({node.private_ip_address}): {results}")
+        # results is of this form:
+        # [{'metric': {}, 'values': [[1749638594.936, '0'], [1749638614.936, '0'], [1749638634.936, '0'], [1749638654.936, '0']]}]
+        return sum(int(v[1]) for v in results[0]['values']) if results else 0
+
+    def test_oos_write(self):
+        """
+        Fill the cluster to 90%
+        Start another write, that would need more space than available
+        Write should fail, but the cluster should not run out of space
+        Scale out the cluster
+        Continue writing data
+        Write should succeed after scale out
+        """
+        self.run_prepare_write_cmd()
+
+        with ignore_stress_errors():
+            result = self.run_write_stress()
+            if result:
+                TestFrameworkEvent(source=self.__class__.__name__,
+                                   message="Writes should have failed, but it succeeded",
+                                   severity=Severity.ERROR).publish()
+
+        self.scale_out()
+
+        self.run_write_stress()
+
+    def test_oos_restart(self):
+        """
+        Fill the cluster to 90%
+        Start another write, that would need more space than available
+        During this write, restart a node
+        The restarted node should not run out of space
+        """
+        self.run_prepare_write_cmd()
+
+        with ignore_stress_errors():
+            stress_thread = Thread(target=self.run_write_stress)
+            stress_thread.start()
+
+            restarted = False
+            while stress_thread.is_alive() and not restarted:
+                for node in self.db_cluster.nodes:
+                    disk_usage = self.get_disk_usage(node)
+                    if disk_usage >= 97:
+                        self.log.info(f"Node {node.name} has reached 97% disk usage, restarting it.")
+                        node.stop_scylla(verify_down=True)
+                        node.start_scylla(verify_up=True)
+                        restarted = True
+                        break
+                sleep(60)
+
+            stress_thread.join()
```

Comment:
It's worth adding a test, that a node was indeed restarted. Otherwise, the test can pass, without testing what was intended.